extensions/amp-story-shopping/0.1/test/test-amp-story-shopping-tag.js

```diff
@@ -65,8 +73,28 @@ describes.realWin(
       storeService.dispatch(Action.ADD_SHOPPING_DATA, shoppingData);
     }
 
-    it('should build and layout shopping tag component', () => {
+    it('should build and layout shopping tag component', async () => {
+      expect(() => shoppingTag.buildCallback()).to.not.throw();
       expect(() => shoppingTag.layoutCallback()).to.not.throw();
+
+      shoppingTag.element.setAttribute('data-product-id', 'sunglasses');
+      await shoppingDataDispatchStoreService();
+      env.sandbox.stub(shoppingTag, 'measureMutateElement').callsFake(() => {
+        expect(shoppingTag.element.textContent).to.be.not.empty;
+      });
+    });
+
+    it('should not build shopping tag if page attachment is missing', async () => {
+      pageEl.removeChild(attachmentElement);
+
+      expect(() => shoppingTag.buildCallback()).to.not.throw();
+      expect(() => shoppingTag.layoutCallback()).to.not.throw();
+
+      shoppingTag.element.setAttribute('data-product-id', 'sunglasses');
+      await shoppingDataDispatchStoreService();
+      env.sandbox.stub(shoppingTag, 'measureMutateElement').callsFake(() => {
```

Comment:
I'm not sure if this stub and the stub in the test above are necessary. Do you know if the tests succeed without it?

Reply:
Yeah, I was able to figure the right way to do it. We should be good to go now.